manager.go

```diff
@@ -92,12 +89,14 @@ type TaskManager struct {
 	min           int
 	workerTimeout time.Duration
 	stopped       atomic.Bool
+
+	logging Logger
 }
 
 // New creates a new taskmanager instance. minCount determines the minimum no of
 // workers and maxCount determines the maximum worker count. timeout is used to determine
 // when workers will be timed out on being idle
-func New(minCount, maxCount int, timeout time.Duration) *TaskManager {
+func New(minCount, maxCount int, timeout time.Duration, log Logger) *TaskManager {
```

Comment:
By default the user should not have to provide anything. So if the log here is nil, we can instantiate a noop logger, which does nothing.